### ISSUE TITLE
Support for MDC when using async logger

### ISSUE THREAD

**Issue by tomaszdomin-arevocom** [mp911de/logstash-gelf#145]
Using log log4j 1.2.17 and logstash-gelf 1.11.2

I've configured gelf appender to work with Async appender:

```
	<appender name="async" class="org.apache.log4j.AsyncAppender">
		<param name="BufferSize" value="500" />
		<appender-ref ref="console" />
		<appender-ref ref="gelf" />
	</appender>
	<root>
		<priority value="info"/>
		<appender-ref ref="async" />
		<!-- for some reason gelf is not handling MDC under async appender -->
	</root>
```
When using such setup MDC is not passed by to gelf appender.
it works fine when I moved gelf to root section.
```
	<root>
		<priority value="info"/>
		<appender-ref ref="async" />
		<appender-ref ref="gelf" />
	</root>
```

I've tried to set on Async Appender level: 
`<param name="LocationInfo" value="true"/>`
It did not help.

**Comment by mp911de** [comment]
Good catch. The MDC isn't fetched from the `LoggingEvent` but statically from the `MDC` class itself. This causes to return always an empty MDC.

**Comment by mp911de** [comment]
That's fixed now, snapshots (1.11.3-SNAPSHOT) are deployed to oss.sonatype.org.
